src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -1438,13 +1484,47 @@ private static DkmClrCompilationResultFlags GetLocalResultFlags(LocalSymbol loca
                             instance);
                     }
 
-                    displayClassVariableNamesInOrder = displayClassVariableNamesInOrderBuilder.ToImmutableAndFree();
                     parameterNames.Free();
                 }
-                else
+            }
+
+            static bool tryAddCapturedPrimaryConstructorParameters(
+                MethodSymbol currentFrame,
+                ImmutableArray<string> shadowingParameterNames,
+                TypeSymbol possiblyCapturingType,
+                (DisplayClassInstance? Instance, ConsList<FieldSymbol> Fields) possiblyCapturingTypeInstance,
```

Comment:
>(DisplayClassInstance? Instance, ConsList<FieldSymbol> Fields) possiblyCapturingTypeInstance [](http://example.com/codeflow?start=16&length=92)

Why a tuple rather than two separate parameters?

Reply:
> Why a tuple rather than two separate parameters?

The values are closely related and I felt the code is easier to follow this way. At least it was easier to write for me.